ggshield/core/git_shell.py

```diff
@@ -76,12 +76,23 @@ def git(
             [_get_git_path()] + command,
             check=check,
             stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
             timeout=timeout,
             env=env,
             cwd=cwd,
         )
         return result.stdout.decode("utf-8", errors="ignore").rstrip()
+    except subprocess.CalledProcessError as e:
+        if "detected dubious ownership in repository" in e.output.decode(
+            "utf-8", errors="ignore"
+        ):
+            raise Exception(
```

Comment:
This causes a massive error message on my machine:

```
$ ggshield secret scan repo .
Traceback (most recent call last):
  File "/home/agateau/src/ggshield/ggshield/core/git_shell.py", line 75, in git
    result = subprocess.run(
  File "/usr/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/bin/git', 'rev-parse', '--show-toplevel']' returned non-zero exit status 128.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/agateau/.local/share/virtualenvs/ggshield-c3rFhSjf/bin/ggshield", line 8, in <module>
    sys.exit(main())
  File "/home/agateau/src/ggshield/ggshield/cmd/main.py", line 143, in main
    return cli.main(args, prog_name="ggshield", standalone_mode=not show_crash_log)
  File "/home/agateau/.local/share/virtualenvs/ggshield-c3rFhSjf/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/agateau/.local/share/virtualenvs/ggshield-c3rFhSjf/lib/python3.10/site-packages/click/core.py", line 1654, in invoke
    super().invoke(ctx)
  File "/home/agateau/.local/share/virtualenvs/ggshield-c3rFhSjf/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/agateau/.local/share/virtualenvs/ggshield-c3rFhSjf/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/agateau/.local/share/virtualenvs/ggshield-c3rFhSjf/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/agateau/src/ggshield/ggshield/cmd/main.py", line 91, in cli
    load_dot_env()
  File "/home/agateau/src/ggshield/ggshield/core/utils.py", line 253, in load_dot_env
    dot_env_path = _find_dot_env()
  File "/home/agateau/src/ggshield/ggshield/core/utils.py", line 238, in _find_dot_env
    if is_git_dir(os.getcwd()):
  File "/home/agateau/src/ggshield/ggshield/core/git_shell.py", line 45, in is_git_dir
    git(["rev-parse", "--show-toplevel"], cwd=wd)
  File "/home/agateau/src/ggshield/ggshield/core/git_shell.py", line 89, in git
    raise Exception(
Exception: Git command failed because of a dubious ownership in repository.
If you still want to run ggshield, make sure you mark the current repository as safe for git.
`git config --global --add safe.repository=<YOUR_REPO>`
```

You can avoid that by raising `UnexpectedError` instead of `Exception` (UnexpectedError comes from our errors.py module).

Reply:
Indeed, this was not really ready for review...  
It should be better now, thanks for your suggestions.